src/AggregateFunctions/AggregateFunctionMLMethod.cpp

```diff
@@ -83,11 +83,11 @@ namespace
                     ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT);
         }
 
-        if (std::is_same<Method, FuncLinearRegression>::value)
+        if (std::is_same_v<Method, FuncLinearRegression>)
         {
             gradient_computer = std::make_unique<LinearRegression>();
         }
-        else if (std::is_same<Method, FuncLogisticRegression>::value)
+        else if (std::is_same_v<Method, FuncLogisticRegression>)
         {
             gradient_computer = std::make_unique<LogisticRegression>();
         }
```

Comment:
I think this should be constexpr check, and in else `static_assert(false, "Such gradient computer is not implemented yet")`.

Can you fix it as part of this PR?

Reply:
Good catch, tnx. Will fix it.

Reply:
Uh ... a static_assert in the else branch would need an ugly workaround (--> https://stackoverflow.com/questions/53945490/how-to-assert-that-a-constexpr-if-else-clause-never-happen). I changed to 2x constexpr checks but would rather keep the current throw in the else. As the else cannot be reached, we wouldn't improve performance anyways.

Reply:
Okay, I found a C++20 solution for the static_assert which looks acceptable imho.